vehicle/carwings/api.go

```diff
@@ -0,0 +1,173 @@
+package carwings
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+)
+
+type API struct {
+	*request.Helper
+	CustomSessionID string
+	tz              string
+	ResultKey       string
+	RefreshTime     time.Time
+}
+
+func NewAPI(log *util.Logger, customSessionID, tz string) *API {
+	return &API{
+		Helper:          request.NewHelper(log),
+		CustomSessionID: customSessionID,
+		tz:              tz,
+	}
+}
+
+func (v *API) Charger(vin string) (ChargerResponse, error) {
+	// api result is stale
+	if v.ResultKey != "" {
+		if err := v.refreshResult(vin); err != nil {
+			return ChargerResponse{}, err
+		}
+	}
+	params := setCommonParams(vin, v.CustomSessionID, v.tz)
```

Comment:
Why is this necessary at all instead of using the carwings library? These functions are already available?